ssl/record/rec_layer_d1.c

```diff
@@ -397,8 +397,12 @@ int dtls1_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
     }
 
     /* Check for timeout */
-    if (dtls1_handle_timeout(s) > 0)
+    if (dtls1_handle_timeout(s) > 0) {
         goto start;
+    } else if (ossl_statem_in_error(s)) {
+        /* dtls1_handle_timeout() has failed with a fatal error */
+        return -1;
+    }
```

Comment:
No reason for the {}

Reply:
Is this a bug fix?

Reply:
Well, dtls1_handle_timeout() can ultimately end up calling dtls1_retransmit_message() which (as a result of this PR) can now call SSLfatal(), so if that happened then we shouldn't continue. So that isn't really a bug fix, its a reaction to a change made in this PR. The fact that it doesn't properly distinguish between 0 and -1 return values from dtls1_handle_timeout() at the moment probably is a bug that should be fixed - but I've not addressed that as part of this PR.

Reply:
With regards to `{}`, that is interesting. I think the current style is that single line statements don't need `{}`. In this case we have two lines - one for a comment, and one for the statement. I have interpreted that as "multi-line" and hence in need of `{}`. Is that wrong? I think I've been doing that for a while.

Reply:
I'm okay with treating commented code as a multi-line block FWIW.

Reply:
For me the {} is when it's needed, and it's only needed when there is more than 1 line code. But I actually prefer to always use {} anyway, so I don't really care.

Reply:
I always add the {}'s even if they are not needed - and definitely would for visual effect if there was a comment. But the coding style is silent on that topic,